teps/0028-task-results-without-results.md

```diff
@@ -0,0 +1,358 @@
+- Very hard to create generic tasks with a common default value.
+
+### Option C
+
+Instead of specifying defaults for task results, leverage task `param` default value. When the task fails and the `param`
+referring to task results has a default, use that `param` default. If that `param` is required and no default specified,
+skip the task.
+
+Pros/Cons:
+
++ This works well without having task results default. The `param` in the task has full control over where its value
+coming from and what is the default if it's not initialized.
++ Works with `continueAfterSkip` feature.
+- Cannot be extended to `when` expressions since task result in `when` expression is not mapped to any `param` in the
+pipeline.
+
+### Option D
+
+No defaults allowed (neither in the `task` producing results nor in the `pipelineTask` for the `task` producing results),
+just skip `finally` task with a reason when a task result is missing.
+
+Pros/Cons:
+
+- Cannot extend this option to `continueAfterSkip` with missing task results.
++ `pipeline` does not have to aware of any defaults or declare any defaults.
+- Change in perception/contract that a `finally` task is being `skipped` where it's supposed to always run. Counter
+argument to this con is, a task result is considered as a variable to a task and if the variable is not initialized,
+`finally` clause throws error and is not executed.
+
+### Option E
+
+`Pipeline` author can check the status of the result producing task and design the `pipeline` to handle success and
+failure scenarios.
+
+Pros/Cons:
+
++ `Pipeline` author has all the freedom to design the `pipeline`.
++ Can be extended to `continueAfterSkip` with missing task results (provided execution status is available within `dag`
+tasks)
+- Cannot work with `when` expressions
+- If `pipeline` author does not design failure scenario, what value should be used for the `param` referring to task
+results?
+
+### Option F
+
+The `cleanup` is necessary when the resources are acquired but not needed if the acquisition fails. This option is
+proposing introducing `when` expression to `finally`. `Cleanup` task in `finally` checks the execution status of
+`acquire` task, and guards `cleanup` if `acquire` failed. This proposal helps avoid dealing with missing task results
+in `finally` i.e. no default or no zero values for task results needed.
+
+Pros/Cons:
+
++ Introducing guarded finally task, its a clean design.
++ `Pipeline` author has full control over its design.
++ Can be extended to `continueAfterSkip`
+- no defaults needed
+- Cannot work with `when` expressions
+
+**Analysis:** We have listed `when` expression with missing task results as one of the possible use case and something
+needs to addressed with the proposal in this TEP. But having difficulty coming up with a possibility of getting into
+`when` expression with missing task results. Let's take an example for the same,
+A(guarded) -> B(continueAfterSkip=true)(guarded with task results from A)  -> C
```

Comment:
```suggestion
A(guarded)(continueAfterSkip=true) -> B(guarded with task results from A)  -> C
```
to say continue executing task `B`, the `continueAfterSkip` should be set to true in guarded task `A`

Reply:
thanks @jerop fixed.